diff --git a/Android.bp b/Android.bp
index 4ee9add..9b1425d 100644
--- a/Android.bp
+++ b/Android.bp
@@ -48,7 +48,7 @@ genrule {
 
 genrule {
     name: "grammar.c",
-    cmd: "M4=$(location m4) $(location bison) -p pcap_ --no-lines --output=$(genDir)/grammar.c $(in)",
+    cmd: "BISON_PKGDATADIR=$$(dirname $$(dirname $$(dirname $(location bison))))/common/bison M4=$(location m4) $(location bison) -p pcap_ --no-lines --output=$(genDir)/grammar.c $(in)",
     out: ["grammar.c"],
     srcs: [":grammar.y"],
     tools: [
@@ -59,7 +59,7 @@ genrule {
     name: "grammar.h",
     // bison --defines outputs an extra file in addition to outputting the parser,
     // so we still use --output to ensure the unused file is still in genDir.
-    cmd: "M4=$(location m4) $(location bison) -p pcap_ --no-lines --defines=$(genDir)/grammar.h --output=$(genDir)/unused $(in)",
+    cmd: "BISON_PKGDATADIR=$$(dirname $$(dirname $$(dirname $(location bison))))/common/bison M4=$(location m4) $(location bison) -p pcap_ --no-lines --defines=$(genDir)/grammar.h --output=$(genDir)/unused $(in)",
     out: ["grammar.h"],
     srcs: [":grammar.y"],
     tools: [
