diff --git a/soong/rustprebuilts.go b/soong/rustprebuilts.go
index 1a0df6213..d1ac6fcaf 100644
--- a/soong/rustprebuilts.go
+++ b/soong/rustprebuilts.go
@@ -127,6 +127,10 @@ func constructLibProps(rlib, solib bool) func(ctx android.LoadHookContext) {
 		if ctx.Config().BuildOS == android.Linux {
 			p.Target.Linux_glibc_x86_64.addPrebuiltToTarget(ctx, name, rustDir, "linux-x86", "x86_64-unknown-linux-gnu", rlib, solib)
 			p.Target.Linux_glibc_x86.addPrebuiltToTarget(ctx, name, rustDir, "linux-x86", "i686-unknown-linux-gnu", rlib, solib)
+			if ctx.Config().UseHostMusl() {
+				p.Target.Linux_musl_x86_64.addPrebuiltToTarget(ctx, name, rustDir, "linux-musl-x86", "x86_64-unknown-linux-musl", rlib, solib)
+				p.Target.Linux_musl_x86.addPrebuiltToTarget(ctx, name, rustDir, "linux-musl-x86", "i686-unknown-linux-musl", rlib, solib)
+			}
 		} else if ctx.Config().BuildOS == android.LinuxMusl {
 			p.Target.Linux_musl_x86_64.addPrebuiltToTarget(ctx, name, rustDir, "linux-musl-x86", "x86_64-unknown-linux-musl", rlib, solib)
 			p.Target.Linux_musl_x86.addPrebuiltToTarget(ctx, name, rustDir, "linux-musl-x86", "i686-unknown-linux-musl", rlib, solib)
@@ -139,7 +143,7 @@ func constructLibProps(rlib, solib bool) func(ctx android.LoadHookContext) {
 }
 
 func rustHostPrebuiltSysrootLibraryFactory() android.Module {
-	module, _ := rust.NewPrebuiltLibrary(android.HostSupportedNoCross)
+	module, _ := rust.NewPrebuiltLibrary(android.HostSupported)
 	android.AddLoadHook(module, constructLibProps( /*rlib=*/ true /*solib=*/, true))
 	return module.Init()
 }
