diff --git a/Android.bp b/Android.bp
index a1e9799..8329f31 100644
--- a/Android.bp
+++ b/Android.bp
@@ -19,8 +19,59 @@ prebuilt_build_tool {
         x86_64: {
             enabled: true,
         },
+        arm64: {
+            enabled: true,
+        },
     },
     target: {
+        linux_arm64: {
+            src: "linux-arm64/bin/javap",
+            deps: [
+                "linux-arm64/lib/libjli.so",
+                "linux-arm64/lib/jrt-fs.jar",
+                "linux-arm64/lib/jvm.cfg",
+                "linux-arm64/lib/server/libjvm.so",
+                "linux-arm64/lib/libverify.so",
+                "linux-arm64/lib/libjava.so",
+                "linux-arm64/lib/libzip.so",
+                "linux-arm64/lib/libjimage.so",
+                "linux-arm64/lib/modules",
+                "linux-arm64/lib/libnio.so",
+                "linux-arm64/lib/libnet.so",
+                "linux-arm64/lib/tzdb.dat",
+                "linux-arm64/lib/libawt.so",
+                "linux-arm64/lib/libawt_headless.so",
+                "linux-arm64/lib/libjavajpeg.so",
+                "linux-arm64/lib/liblcms.so",
+                "linux-arm64/lib/libmanagement.so",
+                "linux-arm64/lib/libmanagement_ext.so",
+                "linux-arm64/conf/security/java.security",
+            ],
+        },
+        linux_musl_arm64: {
+            src: "linux-arm64/bin/javap",
+            deps: [
+                "linux-arm64/lib/libjli.so",
+                "linux-arm64/lib/jrt-fs.jar",
+                "linux-arm64/lib/jvm.cfg",
+                "linux-arm64/lib/server/libjvm.so",
+                "linux-arm64/lib/libverify.so",
+                "linux-arm64/lib/libjava.so",
+                "linux-arm64/lib/libzip.so",
+                "linux-arm64/lib/libjimage.so",
+                "linux-arm64/lib/modules",
+                "linux-arm64/lib/libnio.so",
+                "linux-arm64/lib/libnet.so",
+                "linux-arm64/lib/tzdb.dat",
+                "linux-arm64/lib/libawt.so",
+                "linux-arm64/lib/libawt_headless.so",
+                "linux-arm64/lib/libjavajpeg.so",
+                "linux-arm64/lib/liblcms.so",
+                "linux-arm64/lib/libmanagement.so",
+                "linux-arm64/lib/libmanagement_ext.so",
+                "linux-arm64/conf/security/java.security",
+            ],
+        },
         linux: {
             src: "linux-x86/bin/javap",
             deps: [
