diff --git a/src/client.rs b/src/client.rs
index ab7eb87604..e8b43ac2b9 100644
--- a/src/client.rs
+++ b/src/client.rs
@@ -19,6 +19,8 @@ pub use crate::gpu::do_gpu_display_list;
 #[cfg(feature = "gpu")]
 pub use crate::gpu::do_gpu_display_remove;
 #[cfg(feature = "gpu")]
+pub use crate::gpu::do_gpu_display_resize;
+#[cfg(feature = "gpu")]
 pub use crate::gpu::do_gpu_set_display_mouse_mode;
 #[cfg(feature = "gpu")]
 pub use crate::gpu::ModifyGpuResult;
diff --git a/src/gpu.rs b/src/gpu.rs
index f6eb523811..42e0b9aaf3 100644
--- a/src/gpu.rs
+++ b/src/gpu.rs
@@ -126,6 +126,10 @@ pub enum GpuControlCommand {
     RemoveDisplays {
         display_ids: Vec<u32>,
     },
+    ResizeDisplay {
+        display_id: u32,
+        display: DisplayParameters,
+    },
     SetDisplayMouseMode {
         display_id: u32,
         mouse_mode: MouseMode,
@@ -234,6 +238,20 @@ pub fn do_gpu_display_remove<T: AsRef<Path> + std::fmt::Debug>(
         .into()
 }
 
+pub fn do_gpu_display_resize<T: AsRef<Path> + std::fmt::Debug>(
+    control_socket_path: T,
+    display_id: u32,
+    display: DisplayParameters,
+) -> ModifyGpuResult {
+    let request = VmRequest::GpuCommand(GpuControlCommand::ResizeDisplay {
+        display_id,
+        display,
+    });
+    handle_request(&request, control_socket_path)
+        .map_err(|_| ModifyGpuError::SocketFailed)?
+        .into()
+}
+
 pub fn do_gpu_set_display_mouse_mode<T: AsRef<Path> + std::fmt::Debug>(
     control_socket_path: T,
     display_id: u32,
