diff --git a/src/crosvm/sys/linux/gpu.rs b/src/crosvm/sys/linux/gpu.rs
--- a/src/crosvm/sys/linux/gpu.rs
+++ b/src/crosvm/sys/linux/gpu.rs
@@ -105,4 +105,4 @@
             debug!("gpu fixed blob mapping disabled: not compatible with passthrough GPU.");
             gpu_params.fixed_blob_mapping = false;
-        } else if cfg!(feature = "vulkano") {
+        } else if cfg!(feature = "vulkano") && gpu_params.use_vulkan.unwrap_or(false) {
             // TODO(b/244591751): make fixed_blob_mapping compatible with vulkano for opaque_fd blob
