diff --git a/devices/src/virtio/gpu/virtio_gpu.rs b/devices/src/virtio/gpu/virtio_gpu.rs
--- a/devices/src/virtio/gpu/virtio_gpu.rs
+++ b/devices/src/virtio/gpu/virtio_gpu.rs
@@ -1040,2 +1040,10 @@ impl VirtioGpu {
+        if let Some(shmem_offset) = resource.shmem_offset {
+            if let Some(mapper) = self.mapper.lock().as_mut() {
+                if let Err(e) = mapper.remove_mapping(shmem_offset) {
+                    error!("failed to remove memory mapping: {:#}", e);
+                }
+            }
+        }
+
         if resource.rutabaga_external_mapping {
             self.rutabaga.unmap(resource_id)?;
*** End Patch