diff --git a/disk/src/composite.rs b/disk/src/composite.rs
index 3e7ec8aa58..97b87b5d44 100644
--- a/disk/src/composite.rs
+++ b/disk/src/composite.rs
@@ -24,6 +24,7 @@ use std::sync::atomic::Ordering;
 use std::sync::Arc;
 
 use async_trait::async_trait;
+use base::AsRawDescriptor;
 use base::AsRawDescriptors;
 use base::FileAllocate;
 use base::FileReadWriteAtVolatile;
@@ -397,10 +398,14 @@ impl FileReadWriteAtVolatile for CompositeDiskFile {
 
 impl AsRawDescriptors for CompositeDiskFile {
     fn as_raw_descriptors(&self) -> Vec<RawDescriptor> {
-        self.component_disks
-            .iter()
-            .flat_map(|d| d.file.as_raw_descriptors())
-            .collect()
+        let mut descriptors = vec![self._disk_spec_file.as_raw_descriptor()];
+        descriptors.extend(
+            self.component_disks
+                .iter()
+                .flat_map(|d| d.file.as_raw_descriptors()),
+        );
+
+        descriptors
     }
 }
 
@@ -1069,10 +1074,12 @@ mod tests {
 
     #[test]
     fn triple_file_descriptors() {
+        let spec_file = tempfile().unwrap();
         let file1 = tempfile().unwrap();
         let file2 = tempfile().unwrap();
         let file3 = tempfile().unwrap();
         let mut in_descriptors = vec![
+            spec_file.as_raw_descriptor(),
             file1.as_raw_descriptor(),
             file2.as_raw_descriptor(),
             file3.as_raw_descriptor(),
@@ -1098,7 +1105,8 @@ mod tests {
             file_offset: 0,
             needs_flush: AtomicBool::new(false),
         };
-        let composite = new_from_components(vec![disk_part1, disk_part2, disk_part3]).unwrap();
+        let composite =
+            CompositeDiskFile::new(vec![disk_part1, disk_part2, disk_part3], spec_file).unwrap();
         let mut out_descriptors = composite.as_raw_descriptors();
         out_descriptors.sort_unstable();
         assert_eq!(in_descriptors, out_descriptors);
