diff --git a/host/vulkan/vk_common_operations.cpp b/host/vulkan/vk_common_operations.cpp
--- a/host/vulkan/vk_common_operations.cpp
+++ b/host/vulkan/vk_common_operations.cpp
@@ -736,7 +736,7 @@
         const uint32_t deviceTypeScoreTable[] = {
             100,   // VK_PHYSICAL_DEVICE_TYPE_OTHER = 0,
-            1000,  // VK_PHYSICAL_DEVICE_TYPE_INTEGRATED_GPU = 1,
+            1500,  // VK_PHYSICAL_DEVICE_TYPE_INTEGRATED_GPU = 1,
             2000,  // VK_PHYSICAL_DEVICE_TYPE_DISCRETE_GPU = 2,
             500,   // VK_PHYSICAL_DEVICE_TYPE_VIRTUAL_GPU = 3,
             600,   // VK_PHYSICAL_DEVICE_TYPE_CPU = 4,
         };
